drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -233,6 +233,20 @@ int intel_adsp_hda_dma_status(const struct device *dev, uint32_t channel,
 	stat->pending_length = used;
 	stat->free = unused;
 
+#if CONFIG_SOC_INTEL_ACE20_LNL || CONFIG_SOC_INTEL_ACE30_PTL
+	if (cfg->direction == MEMORY_TO_PERIPHERAL || cfg->direction == PERIPHERAL_TO_MEMORY) {
+		uint32_t llp_l, llp_u, tmp;
+
+		tmp = *DGLLLPL(cfg->base, cfg->regblock_size, channel);
+		llp_u = *DGLLLPU(cfg->base, cfg->regblock_size, channel);
+		llp_l = *DGLLLPL(cfg->base, cfg->regblock_size, channel);
+		if (tmp > llp_l) {
+			/* re-read the LLPU value, as LLPL just wrapped */
+			llp_u = *DGLLLPU(cfg->base, cfg->regblock_size, channel);
+		}
+		stat->total_copied = ((uint64_t)llp_u << 32) | llp_l;
```

Comment:
what would `stat->total_copied` be equal to on other platforms then? Not set at all? Should we set it to 0?

Reply:
Probably, but it is not subject of this PR, I can create a followup to reset the total_copied to 0 (set llp_l/u to 0) if it is not handled and at the same time do similar update to the hd dma.

Reply:
Just a sec, I will update this PR, the gpdma handles this already.